Fix network documentation coverage

jukkar opened the pull request:
This should improve network documentation coverage percentage.
So I added missing doxygen comments when applicable or hide internal documentation which does not need to be documented.
There are lot of commits, but most of them are very small. Each header file changed is placed in its own commit for easier revert / fix if needed.

jukkar committed:
net: doc: net_event.h: Document network events

Add documentation to events that are triggered by the network stack.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: coap: Fix doxygen documentation

Fix doxygen documentation in COAP header files to improve
documentation coverage.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: conn_mgr_connectivity.h: @endcond was incorrectly placed

The net_event_conn_cmd is internal so no need to generate public
documentation for it.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: conn_mgr_monitor.h: Add @file documentation

This improves documentation coverage metric for this file.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: dhcpv4.h: Document DHCPv4 message types

The DHCPv4 message type documentation was missing.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: dummy.h: Document dummy L2 header file

The dummy L2 header file was missing documentation.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: ethernet.h: Document header file properly

Documentation to various public features was missing.
Also some of the public features were marked in hidden section.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: ethernet_bridge.h: Hide internal structs from doxygen

Mark internal structs as hidden to avoid generating public
documentation for it.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: gptp.h: Documentation missing for structs

Add more struct documentation to gptp.h

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: hostname.h: Document hostname length macro properly

Max hostname length was not documented.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: icmp.h: Add ICMPv4/6 packet type documentation

Descriptions of both the ICMPv4 and v6 packet types were missing.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: ieee802154_ie.h: Add more info to documentation

This improves documentation coverage numbers for this file.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: ieee802154_mgmt.h: Improve documentation coverage

Add missing doxygen documentation markers to improve documentation
coverage percentage.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: ieee802154_radio.h: Add more documentation

Added more doxygen comments to ieee802154 radio header file in
order to improve documentation coverage percentage.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: igmp.h: Improve IGMP documentation

Adding doxygen comments to improve documentation coverage.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: ipv4_autoconf.h: Improve documentaion

Add missing doxygen comments to improve documentation coverage.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: net_ip.h: Improve documention

Add doxygen documentation and hide internal documentation from
generated documents.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: net_mgmt.h: Fix net_mgmt documentation

Some documentation was missing in net_mgmt.h header file.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: net_pkt.h: Improve net_pkt documenation

Add more documentation to net_pkt.h

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: net_pkt_filter.h: Improve documentation

Add more documentation to net_pkt_filter.h header file.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: net_stats.h: Improve statistics documentation

Lot of statistics variables were not documented at all.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: net_timeout.h: Fix doxygen comments

Some of the already documented variables were missing proper
doxygen comment.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: offloaded_netdev.h: Fix doxygen comment

The get_type API function was missing doxygen comment.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: phy.h: Add missing documentation

Couple of cabatility checking macros were missing documentation.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: ppp.h: Improve PPP documentation

Some of the symbols in ppp.h were not documented.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: ptp_time.h: Improve documentation

Fix the documentation in ptp_time.h to improve the documentation
coverage.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: sntp.h: Improve documentation

Fix doxygen documentation in sntp.h header file.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: socket.h: Hide internal documentation

Some of the internal macros can be hidden from generated
documentation.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: socket_net_mgmt.h: Hide internal docs

No need to generate documentation for internal data.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: socket_offload.h: Improve documentation

Fix the socket_offload.h header file documentation.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: socket_select.h: Hide internal data from documentation

No need to generate public documentation for internal data.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: socket_types.h: Hide internal data from documentation

No need to generate documentation for internal data.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: socketcan.h: Improve documentation

Add missing doxygen documentation to socketcan.h header file.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: socketutils.h: Add missing @file section

This will improve documentation coverage.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: trickle.h: Add missing doxygen comments

The trickle.h was missing couple of doxygen comments.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: wifi.h: Improve documentation

Add missing doxygen comments and improve documentation coverage.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: wifi_mgmt.h: Improve documentation

Add missing doxygen comments for wifi_mgmt.h header file.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: wifi_nm.h: Hide internal data from documentation

No need to generate documentation for internal data in wifi_nm.h

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: wifi_utils.h: Add missing doxygen documentation

The wifi_utils.h was missing couple of doxygen comments.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: zperf.h: Add missing doxygen comments

The zperp.h header file was missing doxygen comments.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

pdgendt reviewed (commented):
Nice work! There are some compliance/spelling checks that need fixing.

fgrandel reviewed (commented):
Nice! :-) Approved for IEEE 802.15.4 part.

jukkar committed:
net: doc: dns_resolve.h: Improve documentation

Couple of doxygen comments were missing.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: dsa.h: Hide internal documentation

Do not generate documentation for internal symbols.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: ieee802154.h: Improve documentation

Add missing doxygen comments.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: ieee802154_radio_openthread.h: Improve documentation

Add missing doxygen comments.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: lwm2m.h: Improve documentation

Add missing doxygen comments.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: lwm2m_path.h: Add missing @file specifier

This will improve document coverage.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: mdio.h: Add missing doxygen comments

Some defines were missing proper doxygen comment.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: mii.h: Add missing doxygen comments

Some defines were missing doxygen comments.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: mqtt.h: Improve documentation

Some doxygen comments were missing.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: mqtt_sn.h: Reshuffle doxygen comment

Moving doxygen comment to proper place.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: net_context.h: Improve documentation

Some doxygen comments were missing.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: net_if.h: Improve documentation

Add missing doxygen comments.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: openthread.h: Hide internal symbol

No need to show internal symbols in generated documentation.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: socket_service.h: Hide internal documentation

Do not add internal symbols to generated documentation.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: tftp.h: Fix the @file section

The @brief was in wrong place.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: virtual.h: Add missing doxygen comment

The virtual API was not properly documented.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: websocket.h: Add missing doxygen comments

Opcodes were not documented. Also removed init function from
public documentation as it is not needed there.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar commented:
* More CI fixes
* Added more coverage

jukkar committed:
net: doc: buf.h: Improve documentation

Added missing doxygen comments. Also hide internal symbols from
generated documentation.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: canbus.h: Added missing @file section

This will improve documentation coverage.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: http/client.h: Improve documentation

Add missing doxygen comments.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: http/frame.h: Improve documentation

Doxygen documentation was totally missing in this file.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: http/hpack.h: Hide internal documentation

Prevent internal symbols from being in generated documentation.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: http/method.h: Hide internal documentation

Do not generate documentation for internal symbols.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: http/server.h: Improve documentation

Add missing doxygen comments.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: doc: http/service.h: Improve documentation

Add missing doxygen comments.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar commented:
* Fixed HTTP headers as I missed them earlier

rlubos reviewed (dismissed):
Awesome, thanks

jukkar commented:
* More CI fixes

jukkar commented:
With these changes we get pretty good coverage for networking API documentation
https://builds.zephyrproject.io/zephyr/pr/72992/api-coverage/zephyr/net/index.html
https://builds.zephyrproject.io/zephyr/pr/72992/api-coverage/zephyr/net/http/index.html

martinjaeger commented on include/zephyr/net/canbus.h:
```diff
@@ -4,6 +4,10 @@
  */
 
+/** @file
+ *  @brief CANBUS definitions.
```
Only CAN is an abbreviation (for Controller Area Network), so the correct name is CAN bus. We also have the CAN peripheral driver and this header is only used for the Socket API, so I suggest the following change.

```suggestion
 *  @brief CAN bus socket API definitions.
```

glarsennordic reviewed (dismissed):
Looks great, thanks for doing this!

jukkar commented:
* Updated according to comments

martinjaeger reviewed (approved):
Great, thanks!
